src/flask_pyoidc/pyoidc_facade.py

```diff
@@ -238,23 +263,29 @@ def _token_introspection_request(self, access_token: str):
 
         Returns
         -------
-        oic.extension.message.TokenIntrospectionResponse
+        TokenIntrospectionResponse
             Response object contains result of the token introspection.
         """
-        request = {
+        request_args = {
             'token': access_token,
+            'client_id': self._client.client_id,
+            'client_secret': self._client.client_secret,
             'token_type_hint': 'access_token'
         }
-        auth_header = _ClientAuthentication(self._client.client_id, self._client.client_secret)('client_secret_basic',
-                                                                                                request)
         logger.info('making token introspection request')
-        response = self._provider_configuration.requests_session \
-            .post(self._client.introspection_endpoint, data=request, headers=auth_header) \
-            .json()
-
-        return TokenIntrospectionResponse(**response)
-
-    def client_credentials_grant(self, scope: list = None, **kwargs):
+        # Token Introspection is implemented under extension sub-package of
+        # the client in pyoidc.
+        _client_extension = ClientExtension(client_authn_method=CLIENT_AUTHN_METHOD)
```

Comment:
Why recreate this object on every invocation of `_token_introspection_request`?
I'd imagine it more suitable as instance variable (instead of local variable).